src/zkp/pimod.rs

```diff
@@ -372,6 +398,9 @@ fn y_prime_from_y(y: &BigNumber, w: &BigNumber, a: usize, b: usize, N: &BigNumbe
 
 /// Finds unique a,b in {0,1} such that, for y' = (-1)^a * w^b * y, there is an
 /// x such that x^4 = y (mod pq)
+/// In practice, it is sufficient to use only the first element of the
+/// [`Vec<BigNumber>`] as the third output since it is the only part that goes
+/// into the proof.
 #[cfg_attr(feature = "flame_it", flame("PaillierBlumModulusProof"))]
 fn y_prime_combinations(
```

Comment:
I wish this function was better documented with which algorithm that it's using to compute fourth roots. I think cleaning it up is too big for this PR but I might make a separate issue to document this better, take out all the random mutable variables, simplify types (e.g. `a` and `b` are `usize` but they're intended to only be 0 or 1 and it makes `y_prime_from_y` confusing to read because it doesn't make that invariant clear).